mingw-w64-git/git-wrapper.c

```diff
@@ -607,6 +607,46 @@ static void initialize_top_level_path(LPWSTR top_level_path, LPWSTR exepath,
 	}
 }
 
+static void maybe_read_config(LPWSTR top_level_path)
+{
+	WCHAR config_path[MAX_PATH];
+	char line[1024];
+	FILE *f;
+
+	wcscpy(config_path, top_level_path);
+	wcsncat(config_path, L"\\etc\\git-bash.config", MAX_PATH - wcslen(top_level_path) - 1);
```

Comment:
I assume `top_level_path` never has a trailing `\`?
What happens if `wcslen(top_level_path)` is just shy of `MAX_PATH`?

Reply:
> I assume `top_level_path` never has a trailing `\`?

Does it matter? It will have a double backslash, which is fine. But yes, I think I trim the trailing backslash.

From reading over the code, it does looks like `top_level_path` never has a trailing backslash... it uses `PathRemoveFileSpec()` to trim the path.

>  What happens if `wcslen(top_level_path)` is just shy of `MAX_PATH`?

Then the path will be very unlikely to refer to the correct file, and most likely fail to open anything. However, keep in mind that `top_level_path` is determined by appending `mingw64\bin` and seeing whether that exists (and falling back to looking for `mingw\bin`). So that path must fit inside `MAX_PATH` already, I think. See https://github.com/git-for-windows/MINGW-packages/blob/b1fdea9f6f7b20f46779fe4e1e3b772de9b3a6ce/mingw-w64-git/git-wrapper.c#L570-L608 for details.

To be honest, I am not all too worried about `MAX_PATH` in the Git wrapper, as it is unlikely that `mingw64\libexec\git-core\git-remote-https.exe`  would work, so whoever installs Git into such a long path _will_ be in for a world of pain. And there is pretty little I can (or want to) do about it 😀